pkg/nsm/apis/internsm/internsm.proto

```diff
@@ -0,0 +1,105 @@
+// Note: A connection in Requested state may have a partially specified RemoteConnectionId.  A RemoteConnectionId must be fully specified in CONNECTED state
+message RemoteConnectionId {
+    // source_nsm_id - Id of the source NSM - provided authoritatively by source NSM in RemoteConnectionRequest
+    // Can be used for 'recovery' in the event of an NSM restart by 'Monitoring' all RemoteConnections
+    // from its peers matching its own source_nsm_id
+    string source_nsm_id = 1;
+    // destination_nsm_id - Id of the destination NSM - provided authoritatively by the destination NSM in RemoteConnectionReply
+    string destination_nsm_id = 2;
+    // source_connection_id - Id of the source NSM's end of the connection - provided authoritatively by the source NSM in RemoteConnectionRequest
+    string source_connection_id = 3;
+    // destination_connection_id - Id of the destination NSM's end of the connection - provided authoritatively by the destination NSM in the RemoteConnectionReply
+    string destination_connection_id = 4;
+}
+
+// RemoteConnectionState - state of the RemoteConnection
+enum RemoteConnectionState {
+    // REQUESTED - State of RemoteConnection in RemoteConnectionRequest
+    REQUESTED = 0;
+    // CONNECTED - State of RemoteConnection in RemoteConnectionReply in response to RequestRemoteConnection
+    CONNECTED = 1;
+    // CLOSE_REQUESTED - State of RemoteConnection passed to RemoteConnectionReply in response to DeleteRemoeConnection
+    CLOSE_REQUESTED = 2;
+    // CLOSED - State of the RemoteConnection in RemoteConnectionReply in response to DeleteRemoteConnection
+    CLOSED = 3;
+}
+
+// RemoteConnection - represents a RemoteConnection, may be partially or fully specified
+message RemoteConnection {
+    //  id of the remote connection - must be unique when fully specified
+    RemoteConnectionId id = 1;
+    //  state of the remote connection
+    RemoteConnectionState state = 2;
+    //  network_service_name - Name of the Network Service at the destination of the RemoteConnection
+    string network_service_name = 3;
+    //  network_service_endpoint_name - Name of the specific Network Service Endpoint at the destination of the RemoteConnection
+    string network_service_endpoint_name = 4;
+    //  remote_mechanism - RemoteMechanism for the remote connection.  Must be empty in RequestRemoteConnection.  Must by fully specified in RemoteConnectionReply
+    common.RemoteMechanism remote_mechanism = 5;
+    // connection_context - conntection_context of the remote connection.
+    common.ConnectionContext  connection_context = 6;
+    // labels - meta data about the RemoteConnection.  Typically provided in RequestRemoteConnection and not changed from there.
+    map<string,string> labels = 7;
+}
+
+// RemoteConnectionRequest - Request for a RemoteConnection
+message RemoteConnectionRequest {
+   // remote_connection - partially specified RemoteConnection being requested
+   RemoteConnection remote_connection = 1;
```

Comment:
There is already common.RemoteMechanism in RemoteConnection message, see comment above for consolidating  messages into RemoteConnection.

Reply:
@sbezverk See previous comment about a RemoteConnection having precisely one RemoteMechanism... a *request* may contain a preference ordered list of options, but a RemoteConnection never has more than one.